src/pudl/extract/ferc1.py

```diff
@@ -903,3 +912,95 @@ def extract_xbrl_metadata(
                 xbrl_meta_out[pudl_table][period] = []
 
     return xbrl_meta_out
+
+
+# DAGSTER ASSETS
+def create_raw_ferc1_assets() -> list[SourceAsset]:
+    """Create SourceAssets for raw ferc1 tables.
+
+    SourceAssets allow you to access assets that are generated elsewhere.
+    In our case, the xbrl and dbf database are created in a separate dagster repository.
+
+    Returns:
+        A list of ferc1 SourceAssets.
+    """
+    # Deduplicate the table names because f1_elctrc_erg_acct feeds into multiple pudl tables.
+    dbf_table_names = tuple({v["dbf"] for v in TABLE_NAME_MAP.values()})
+    raw_ferc1_dbf_assets = [
+        SourceAsset(
+            key=AssetKey(table_name), io_manager_key="ferc1_dbf_sqlite_io_manager"
+        )
+        for table_name in dbf_table_names
+    ]
```

Comment:
Hmm my use of `TABLE_NAME_MAP` felt a little janky. Currently, the `TABLE_NAME_MAP` has two purposes in the dagster ETL:
1. Maintain a list of raw FERC tables so we can create `SourceAssets` for them. The source assets are created by the `pudl.extract.ferc1.create_raw_ferc1_assets()` function. I'm doing some weird stuff like adding `_duration` and `_instant` suffixes back and deduplicating raw tables that are fed into multiple clean tables.
2. Specify the raw tables the clean tablers depend on. The `pudl.transform.ferc1.ferc1_transform_asset_factory()` function accesses the `TABLE_NAME_MAP` object to programmatically create the dependencies.

A simpler approach might be to remove the `TABLE_NAME_MAP` object and replace it with:
1. A list of raw table names we need to create `SourceAssets` for.
2. Manually specify the dependencies in dagster assets instead of programmatically creating the assets using the `TABLE_NAME_MAP`.

Reply:
@cmgosnell what do you think is best?

Reply:
hm okay let me make sure i understand the nuances here. We're using the `TABLE_NAME_MAP` right now both in the extract and transform steps. And changing one has some dependencies on the other? We could just make a list of raw tables and make lil wrapper `asset` functions like you did for the `plants_steam_ferc1` table. The **all asset transform method**. OR the **factory transform method**.... could make the factory in such a way where we could add any number of tables? Like could `ferc1_transform_asset` take a list of assets instead of just one? based on `TABLE_NAME_MAP` or some otherwise compiled list of dependencies?

I like the factory generally. It seems to enable a pretty condensed way to call most of the transformers. It feels almost silly to have a factory and then have a few tables that need standalone assets (right now only steam but austen also added the `electricity_sales_by_rate_schedule_ferc1` table, which takes a bazillion xbrl tables). The nice thing about the asset-only transform-ness is that it is straightforward! So if your leaning in that direction I think that makes sense.